Exercise_2.cpp

```diff
@@ -1,47 +1,81 @@
-#include <bits/stdc++.h> 
-using namespace std;  
+// Time Complexity - I was unable to visualize what the time comlexity would be here for partition() and the quicksort().
+// Problems faced - Sometimes my program fails to build. I tried to figure it out but could not understand why it was happening.
+// I was getitng a stack buffer underflow error.
+
+#include <bits/stdc++.h>
+using namespace std;
   
-// A utility function to swap two elements  
-void swap(int* a, int* b)  
-{  
-    //Your Code here 
-}  
+// A utility function to swap two elements
+void swap(int* a, int* b)
+{
+    //Your Code here
+    int temp;
+    temp = *a;
+    *a = *b;
+    *b = temp;
+}
   
-/* This function takes last element as pivot, places  
-the pivot element at its correct position in sorted  
-array, and places all smaller (smaller than pivot)  
-to left of pivot and all greater elements to right  
+/* This function takes last element as pivot, places
+the pivot element at its correct position in sorted
+array, and places all smaller (smaller than pivot)
+to left of pivot and all greater elements to right
 of pivot */
-int partition (int arr[], int low, int high)  
-{  
-    //Your Code here 
-}  
+int partition (int arr[], int low, int high)

```

Comment:
int partition(int arr[], int low, int high) 
    { 
   	//Write code here for Partition and Swap 
        int pivot = arr[high];  
        int i = (low-1);  
        for (int j=low; j<high; j++) 
        { 
            if (arr[j] < pivot) 
            { 
                i++; 
               swap(arr,i,j);

            } 
        } 

        // swap arr[i+1] and arr[high] (or pivot) 
        swap(arr,i+1,high);

        return i+1; 

    }